src/config/plugin.ts

```diff
@@ -169,23 +170,24 @@ export class Plugin implements IPlugin {
     return ids
   }
 
-  findCommand(id: string, opts: {must: true}): Command.Class
+  async findCommand(id: string, opts: {must: true}): Promise<Command.Class>
 
-  findCommand(id: string, opts?: {must: boolean}): Command.Class | undefined
+  async findCommand(id: string, opts?: {must: boolean}): Promise<Command.Class | undefined>
 
-  findCommand(id: string, opts: {must?: boolean} = {}): Command.Class | undefined {
-    const fetch = () => {
+  async findCommand(id: string, opts: {must?: boolean} = {}): Promise<Command.Class | undefined> {
+    const fetch = async () => {
       if (!this.commandsDir) return
       const search = (cmd: any) => {
         if (typeof cmd.run === 'function') return cmd
         if (cmd.default && cmd.default.run) return cmd.default
         return Object.values(cmd).find((cmd: any) => typeof cmd.run === 'function')
       }
+      const isESM = this.pjson.type === 'module'
       const p = require.resolve(path.join(this.commandsDir, ...id.split(':')))
-      this._debug('require', p)
+      this._debug(isESM ? '(import)' : '(require)', p)
       let m
       try {
-        m = require(p)
+        m = isESM ? await ModuleLoader.importDynamic(p) : require(p)
```

Comment:
Can you call `ModuleLoader.load(this, p)` since that has the importDynamic/require logic?

Reply:
Both load and loadWithData pass the string through ModuleLoader.resolvePaths

This require.resolve line  (`require.resolve(path.join(this.commandsDir, ...id.split(':')))`) uses `this.commandDir`
https://github.com/oclif/core/pull/160/files/28566d746b4ec5b0d8be3ef2ba5d7808d213452d#diff-0edcef2f1da996cde64784a30d76043fbe58fb9352f6b0189e1d135b868ee8f1L184

In ModuleLoader.resolvePaths `config.root` is used:
`filePath = isESM ? require.resolve(path.join(config.root, modulePath))`

---------

However this could be a solution (tests pass with this below):
```js
      let m
      try {
        const p = path.join(this.pjson.oclif.commands as string, ...id.split(':'))
        const {isESM, module, filePath} = await ModuleLoader.loadWithData(this, p)
        this._debug(isESM ? '(import)' : '(require)', filePath)
        m = module
      } catch (error) {
        if (!opts.must && error.code === 'MODULE_NOT_FOUND') return
        throw error
      }

```

Reply:
A further note is that since `error.code` is being checked in the catch statement that it should be verified that the code is the same for require and dynamic import. My assumption is that may not be the case.

Reply:
@amphro It turns out in a CommonJS environment that both require and import dynamic return an error code of `MODULE_NOT_FOUND` so that part is fine.  

It is interesting to note that in an ESM environment import dynamic has an error code of `ERR_MODULE_NOT_FOUND` and require through module.createRequire returns `MODULE_NOT_FOUND`.

Reply:
I have to say I'm certainly shying away from any custom error wrapping of the loading process. `MODULE_NOT_FOUND`/ `ERR_MODULE_NOT_FOUND` is one of many errors that can be thrown; IE various errors in the code being loading itself should be posted.

So, I have updated the PR with what I think is best. 

For reference I have implemented my own dual Node / browser module loading code as an ESM Node module that is similar to what is being discussed here (has test cases / test framework):
https://github.com/typhonjs-node-utils/loader-module